lib/layoutTestingMain.js

```diff
@@ -0,0 +1,56 @@
+var layoutEngine = require('./layout/forceDirectedLayout');
+var parse = require('./parse/parse');
+
+var log = function(message) {
+  console.log(message);
+
+  var li = document.createElement('li');
+  li.innerHTML = message;
+  document.getElementById('output').appendChild(li);
+};
+
+var StopWatchLogger = function() {};
+StopWatchLogger.prototype = {
+  start: function(name) {
+    log('' + name + ' started');
+    this.name = name;
+    this.startTime = Date.now();
+  },
+
+  stop: function() {
+    log('' + this.name + ' completed [' + (Date.now() - this.startTime) + 'ms]');
+  }
+};
+
+function processGcl(gcl) {
+  var stopwatch = new StopWatchLogger();
+
+  stopwatch.start('Parsing');
+  var parseResult = parse(gcl);
+  stopwatch.stop();
+
+  var errors = parseResult.gclErrors;
+  console.log(errors);
+
+  var model = parseResult.model;
+
+  stopwatch.start('Layout engine');
+  layoutEngine(model);
+  stopwatch.stop();
+  console.debug(model);
+}
+
+gclSubmit = function() {
```

Comment:
So I have this function as the onsubmit for a form. This seemed to be the only way to make it available from the HTML. I'd rather bind the form event here, but I was having trouble preventing it from being submitted that way. Suggestions?